odoo/orm/domains.py

```diff
@@ -1111,6 +1111,22 @@ def _optimize_in_required(condition, model):
     return DomainCondition(condition.field_expr, condition.operator, value)
 
 
+@operator_optimization(['any'], OptimizationLevel.FULL)
+def _optimize_bypass_comodel_id_lookup(condition, model):
+    """Avoid comodel's subquery, if it can be compared with the field directly"""
+    field = condition._field(model)
+    if (
+        model.env.is_superuser()
+        and field.relational
+        and not field.auto_join
+        and isinstance(subdomain := condition.value, DomainCondition)
+        and subdomain.field_expr == 'id'
+        and subdomain.operator in ('in', 'not in', 'any', 'not any')
```

Comment:
```suggestion
        and subdomain.operator in ('in', 'not in')
```
There is already an optimization that transforms `('id', 'any', D)` into D